### PR TITLE
[ANCHOR-371] Refactor metrics report

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

- Refactor the metrics
- Add metrics to sep, observer, platform, and event servers.
- Disable MetricEmitterService which should belong to BI (Business Intelligence)

### Why

Improve the metric reports.

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/util/MetricConstants.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  // payment observer metrics
+  public static final String PAYMENT_OBSERVER_LATEST_BLOCK_READ =
+      "payment_observer.latest_block_read";
+  public static final String PAYMENT_OBSERVER_LATEST_BLOCK_PROCESSED =
+      "payment_observer.latest_block_processed";
+
+  // event processor metrics
+  public static final String EVENT_RECEIVED = "event_processor.event_received";
+  public static final String EVENT_PROCESSED = "event_processor.event_processed";
+  public static final String TV_BUSINESS_SERVER_CALLBACK = "business_server_callback_api";
```
Comment (philipliu) [inline core/src/main/java/org/stellar/anchor/util/MetricConstants.java:47]: Question: What does the `TV` prefix mean?

**Review by philipliu — APPROVED** [review]
Minor comments but LGTM overall.

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java [hunk 2]
```diff
@@ -158,6 +162,8 @@ public ChallengeResponse createChallenge(ChallengeRequest challengeRequest)
       ChallengeResponse challengeResponse =
           ChallengeResponse.of(txn.toEnvelopeXdrBase64(), appConfig.getStellarNetworkPassphrase());
       trace("challengeResponse:", challengeResponse);
+      // increment counter
+      sep10ChallengeCreatedCounter.increment();
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java:166]: I think this is a good idea to annotate functions. I am not sure the possibility of annotating a block of code. 

Due to the time pressure of the project, I will have to table the suggestion for future improvements.

File: platform/src/main/java/org/stellar/anchor/platform/service/TransactionService.java [hunk 3]
```diff
@@ -88,29 +112,52 @@ public TransactionService(
...
-    if (sep == TransactionsSeps.SEP_31) {
-      txn = txn31Store.findTransactions(params);
-    } else if (sep == TransactionsSeps.SEP_24) {
-      txn = txn24Store.findTransactions(params);
-    } else {
-      throw new BadRequestException("SEP not supported");
+    switch (sep) {
+      case SEP_31:
+        txn = txn31Store.findTransactions(params);
+        findSep6TransactionsCounter.increment();
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/service/TransactionService.java:149]: Fixed. Thank you.

**Pull request merged**